src/components/policy/src/policy/src/policy_manager_impl.cc

```diff
@@ -959,5 +975,48 @@ void PolicyManagerImpl::set_cache_manager(
   cache_ = cache_manager;
 }
 
-}  //  namespace policy
+// Implementation RetrySequenceTimewatch
+
+PolicyManagerImpl::RetrySequenceTimewatch::RetrySequenceTimewatch(
+    PolicyManagerImpl* ptr)
+    : timer_retry_sequence_(
+          "Retry sequence timer", this, &RetrySequenceTimewatch::Retry, true) {
+  DCHECK(ptr);
+  policy_manager_ = ptr;
+}
+
+PolicyManagerImpl::RetrySequenceTimewatch::~RetrySequenceTimewatch() {
+  timer_retry_sequence_.stop();
+}
 
+void PolicyManagerImpl::RetrySequenceTimewatch::Retry() {
+  DCHECK(policy_manager_);
+
+  policy_manager_->RequestPTUpdate();
+
+  std::uint32_t timeout = policy_manager_->NextRetryTimeout();
```

Comment:
@AGritsevich If you decided to use uint32_t without std please keep consistency - so remove it here also like you did in other places.